pkg/apply/applier.go

```diff
@@ -140,15 +142,19 @@ func (a *Applier) Run(ctx context.Context, invInfo inventory.InventoryInfo, obje
 			return
 		}
 
-		// Validate the resources to make sure we catch those problems early
-		// before anything has been updated in the cluster.
-		if err := (&object.Validator{
-			Mapper: mapper,
-		}).Validate(objects); err != nil {
-			handleError(eventChannel, err)
-			return
+		// Validate objects
+		invalidObjects, err := validateObjects(objects, mapper, eventChannel)
+		if err != nil {
+			klog.V(4).Infof("calculated %d invalid objs", len(invalidObjects))
+			if options.ValidationPolicy == validator.ExitEarly {
```

Comment:
It seems like errors in name and/or namespace would be somewhat different than other validation errors. They will have to be detected before we update the inventory. Skipping those means we would make sure they are not added to the inventory, but also make sure we don't end up accidentally pruning any resources that have been skipped.

Reply:
AFAIK, skipped objects aren't added or removed from the inventory. If they were previously in the inventory, I think they're retained, right?

So is there a problem?

Reply:
I've added invalid objects to the TaskContext now. Invalid objects will only be added to the inventory IFF they already existed in the inventory. So that can only really happen for annotations that were changed after being applied, and not for missing name/kind/namespace.